go/interpreter/lfvm/interpreter_intergration_test.go

```diff
@@ -0,0 +1,689 @@
+				}
+				if want, got := expectedStatus, ctxt.status; want != got {
+					t.Errorf("execution failed: status is %v, wanted %v", got, want)
+				}
+
+				expectedPC := int32(len(test.code))
+				if test.pcAfter != nil {
+					expectedPC = int32(*test.pcAfter)
+				}
+				if want, got := expectedPC, ctxt.pc; want != got {
+					t.Errorf("execution failed: pc is %v, wanted %v", got, want)
+				}
+
+				if want, got := test.stack.sizeAfter, ctxt.stack.len(); want != got {
+					t.Errorf("execution failed: stack size is %v, wanted %v", got, want)
+				}
+
+				// Check gas consumption
+				if want, got := test.gas.after, ctxt.gas; want != got {
+					t.Errorf("execution failed: gas consumption is %v, wanted %v",
+						test.gas.before-got,
+						test.gas.before-want)
+				}
+
+				// Check gas refund
+				if want, got := test.gasRefund, ctxt.refund; want != got {
+					t.Errorf("execution failed: gas refund is %v, wanted %v", got, want)
+				}
+			})
+		})
+	}
+}
+
+////////////////////////////////////////////////////////////////////////////////
+// Helper utils
+
+// testParameter is a tool that allows to define before-after pairs of values for a test.
+type testParameter[T any] struct {
+	before, after T
+}
+
+type gas tosca.Gas
+type pcOverride int32
+
+// overridePc will mark the tests to check for a specific PC value.
+// When not used, len(code) is used as the expected PC value.
+func overridePc(pc int32) *pcOverride {
+	res := pcOverride(pc)
+	return &res
+}
+
+// revisionConstraint is a helper struct to define a revision constraint for a test.
+// revisions constraints are defined by the [min, max) range.
+// zero value means no constraint.
+type revisionConstraint struct {
+	min, max *tosca.Revision
+}
+
+// isValidFor checks if the constraint is valid for revision.
+func (rc revisionConstraint) isValidFor(revision tosca.Revision) bool {
+	if rc.min != nil && revision < *rc.min {
+		return false
+	}
+	if rc.max != nil && revision >= *rc.max {
+		return false
+	}
+	return true
+}
+
+// validBefore creates a revision constraint that is valid for revisions greater equal than argument.
+func validFrom(revision tosca.Revision) revisionConstraint {
+	return revisionConstraint{min: &revision}
+}
+
+// validBefore creates a revision constraint that is valid for revisions older than argument.
+func validBefore(revision tosca.Revision) revisionConstraint {
+	return revisionConstraint{max: &revision}
+}
+
+// validInRange creates a revision constraint that is valid only for revision.
+func validOnlyIn(revision tosca.Revision) revisionConstraint {
+	next := revision + 1
+	return revisionConstraint{min: &revision, max: &next}
+}
+
+// validInRange creates a revision constraint that is valid in the range [min, max]
+func validInRange(min, max tosca.Revision) revisionConstraint {
+	next := max + 1
+	return revisionConstraint{min: &min, max: &next}
+}
+
+// generateOpCodesInRange generates a slice of opcodes in the range [start, end].
```

Comment:
```suggestion
// getOpCodesInRange generates a slice of opcodes in the range [start, end].
```

Reply:
I rather not call functions which are not a getter get, this is a generator. mabe make...

Reply:
nevermind, it seems that i remaned it and you are right. the comment was out of date